reward/src/lib.rs

```diff
@@ -0,0 +1,534 @@
+            }
+        }
+
+        Ok((
+            Options {
+                range,
+                dry,
+                amount,
+                currency,
+                strategy,
+                provider,
+                signer,
+            },
+            vec![],
+        ))
+    }
+}
+
+pub fn run(options: Options) -> anyhow::Result<()> {
+    let rt = tokio::runtime::Runtime::new()?;
+    let provider = ethereum::provider(options.provider)?;
+    let signer_opts = options.signer;
+    let (wallet, provider) = rt.block_on(term::ethereum::get_wallet(signer_opts, provider))?;
+    let signer: Arc<_> = SignerMiddleware::new(provider, wallet).into();
+    let profile = profile::default()?;
+    let storage = profile::read_only(&profile)?;
+    let repo = git::Repository::open(Path::new("."))?;
+
+    let mut revwalk = repo.revwalk()?;
+    let range = if options.range.is_none() {
+        if !repo.tag_names(Some("rad/reward"))?.is_empty() {
+            Ok("rad/reward..".to_string())
+        } else {
+            Err(anyhow!("You haven't supplied a range of commits"))
+        }
+    } else {
+        options.range.ok_or_else(|| anyhow!(""))
+    }?;
+    revwalk.push_range(&range)?;
+
+    // from identities in local monorepo build a map of: fp -> ens
+    let mut ssh_to_ens: HashMap<String, _> = HashMap::new();
+    identities::any::list(&storage)?
+        .filter_map(|identity| {
+            if let Ok(Person(person)) = identity {
+                let peer_ids = person
+                    .delegations()
+                    .iter()
+                    .map(|key| {
+                        let peer_id: PeerId = key.to_owned().into();
+                        peer_id
+                    })
+                    .collect::<Vec<PeerId>>();
+                if !peer_ids.is_empty() {
+                    Some((keys::to_ssh_fingerprint(&peer_ids[0]).ok()?, person))
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<(_, _)>>()
+        .iter()
+        .map(|(fp, doc)| {
+            if let Ok(Some(ens)) = doc.payload().get_ext::<radicle_common::person::Ens>() {
+                ssh_to_ens.insert(fp.to_string(), ens);
+            }
+        })
+        .for_each(drop);
+
+    term::headline("Retrieving commits...");
+    let mut table = term::Table::default();
+    table.push([
+        term::format::bold("Commit"),
+        term::format::bold("Signature"),
+        term::format::bold("ENS"),
+    ]);
+
+    let mut head_sha1 = None;
+    // show commits in provided range and collect all ens which have contributed
+    let all_ens = revwalk
+        .filter_map(|sha1| {
+            let sha1 = match sha1 {
+                Err(_) => return None,
+                Ok(sha1) => sha1,
+            };
+            if head_sha1.is_none() {
+                head_sha1 = Some(sha1);
```

Comment:
This is a roundabout way to get the head, maybe just ust `repo.head()` where you need it

Reply:
It's not always `repo.head()`, if user specifies a specific range, HEAD of that range could be different from repo's head.

Reply:
Yes, you're right!

Reply:
Yes, you're right!